cmd/buildah/build.go

```diff
@@ -371,6 +378,7 @@ func buildCmd(c *cobra.Command, inputArgs []string, iopts buildOptions) error {
 		OS:                      systemContext.OSChoice,
 		Out:                     stdout,
 		Output:                  output,
+		BuildOutput:             iopts.BuildOutput,
```

Comment:
Doing the parsing in `GetBuildOutput` above is nice; actually providing the result of the parsing here, instead of forcing the caller to parse again, would be cleaner. It avoids unreachable error cases in the ultimate implementation, and makes it much less likely to confuse the string syntax and the parsed data.

Reply:
I do think that all such tools should do their own parsing; that way they will fail quickly on invalid input, rather than triggering the long and costly build, and only failing in the commit step. So if they do the parsing, they might just as well provide the parsed value, and that eliminates a failure case in the commit step.

Also consider the podman-remote case, where the remote server basically ignores the users’ input and needs to provide a `BuildOutput` value pointing at a temporary tarball location/stream. If the options allow parsing a structure, the server can just pass a `BuildOutputOption{Path:theTemporaryFile, …}`. If the options require a string, the server will need to, quite pointlessly, format one.

This is not blocking from my perspective, and there might be something that makes passing the deconstructed value unattractive. Still, I have a preference :)

Reply:
> I do think that all such tools should do their own parsing

That implies that `GetBuildOutput` would have to be in an externally-importable package, not in `buildah/internal`.

Reply:
moved this to external parse pkg so it can be used by external parties if needed.